hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestEditLogTailer.java

```diff
@@ -488,4 +509,19 @@ private static MiniDFSCluster createMiniDFSCluster(Configuration conf,
         .build();
     return cluster;
   }
+
+  private static final class TestTimer extends Timer {
```

Comment:
I think we already have a class called `ManualTimer` or `FakeTimer`, can you check again?

Reply:
Yes we have `FakeTimer`, however, it's c'tor allows only fixed values initially, which can be incremented later (IMHO, not efficient for our test):
```
  public FakeTimer() {
    // Initialize with a non-trivial value.
    now = 1577836800000L; // 2020-01-01 00:00:00,000+0000
    nowNanos = TimeUnit.MILLISECONDS.toNanos(1000);
  }
```

Reply:
I think the common faker timer would be better if we create a new public constructor? Using `advance` would be more clearer when reading.
```
FakeTimer fakeTimer = new FakeTimer(standby.getNamesystem().getEditLogTailer().getTimer().monotonicNow());
standby.getNamesystem().getEditLogTailer().setTimerForTest(fakerTimer);
...
fakeTimer.advance(inSufficientTimeForLogRoll);
try {
    checkForLogRoll(active, origTxId, noLogRollWaitTime);
    fail("Expected to timeout");
}
....
fakeTimer.advance(sufficientTimeForLogRoll);
checkForLogRoll(active, origTxId, logRollWaitTime);
```